openfl-tutorials/interactive_api/Tensorflow_CIFAR_tfdata/envoy/cifar10_shard_descriptor.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""CIFAR10 Shard Descriptor (using `tf.data.Dataset` API)"""
+import logging
+from typing import List, Tuple
+
+import tensorflow as tf
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+
+logger = logging.getLogger(__name__)
+
+
+class CIFAR10ShardDescriptor(ShardDescriptor):
```

Comment:
Could we split up this class into ShardDescriptor and ShardDataset?

Reply:
I think this is really a question of whether you want to adhere to the ShardDescriptor and ShardDataset design. IMHO, it doesn't make sense in this case, unless you want to enforce it strictly.